### PR TITLE
Om dsl

### PR DESCRIPTION
**Pull request by chrisseaton**
Sorry this is a repeat of #40, which failed on other test systems. It turns out that my implementation was dependent on the order in which the compiler passed methods to my annotation processor. I've fixed that.

### CODE REVIEW

File: truffle/com.oracle.truffle.api.object.dsl/src/com/oracle/truffle/api/object/dsl/Layout.java [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+ *
+ * <h2>Custom Identifiers</h2>
+ *
+ * By default, internal {@link HiddenKey} identifiers with descriptive names will be created for
+ * your properties automatically. You can also specify the identifiers to use by defining them as
+ * public static final fields in the interface.
+ *
+ * {@codesnippet customid}
+ *
+ * @since 0.8 or earlier
```
Comment (woess) [inline truffle/com.oracle.truffle.api.object.dsl/src/com/oracle/truffle/api/object/dsl/Layout.java]: should be `@since 0.12` (+2 other occurrences)

Comment (chrisseaton) [inline truffle/com.oracle.truffle.api.object.dsl/src/com/oracle/truffle/api/object/dsl/Layout.java]: Ah yes of course - thanks - fixed.

**Comment by woess** [comment]
Nitpicking: the commits and their order are a bit confusing. I would have expected a 1 commit that is the "unrevert" followed by another one that is the bug fix.

**Comment by chrisseaton** [comment]
@woess done

**Comment by woess** [comment]
thanks

**Pull request merged**